catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/IngestCommand.java

```diff
@@ -329,6 +359,78 @@ private CreateResponse createMetacards(CatalogFacade catalog, List<Metacard> lis
         return catalog.create(createRequest);
     }
 
+    private void createContentItem(Metacard metacard) {
+        List<File> fileList = fileMap.get(metacard.getId());
+
+        if (fileList != null) {
+            List<ContentItem> contentItemList = new ArrayList<>();
+            ContentItem contentItem;
+
+            for (File file : fileList) {
+                ByteSource byteSource = com.google.common.io.Files.asByteSource(file);
+                // The fileName is checked earlier and is guaranteed to follow correct convention
+                String fileName = file.getName()
+                        .split("-")[1];
+
+                String fragment = null;
+                if (!file.getPath()
+                        .contains(CONTENT_PATH + metacard.getId())) {
+                    fragment = StringUtils.substringBetween(file.getPath(),
+                            CONTENT_PATH,
+                            "/" + metacard.getId());
+                }
+                contentItem = new ContentItemImpl(metacard.getId(),
+                        fragment,
+                        byteSource,
+                        metacard.getContentTypeName(),
+                        fileName,
+                        file.length(),
+                        metacard);
+                contentItemList.add(contentItem);
+            }
+            addContentItemToStore(contentItemList, metacard.getId());
+        }
+    }
+
+    private void addContentItemToStore(List<ContentItem> contentItems, String metacardId) {
+        CreateStorageRequest createStorageRequest = new CreateStorageRequestImpl(contentItems,
+                metacardId,
+                new HashMap<>());
+        StorageProvider storageProvider = getStorageProvider();
```

Comment:
You should not be directly accessing the storage provider. See my comment below.